utils/logging.js

```diff
@@ -33,14 +33,24 @@ const prettyJSON = (() => {
       .join(" ");
 })();
 
+const $try = (fn, defaultValue = null) => {
+  try {
+    return fn();
+  } catch {
+    return defaultValue;
+  }
+};
+
 /** @type {import('morgan').FormatFn} */
 const formatter = (tokens, req, res) => {
   const date = tokens.date(req, res, "iso");
   const remoteAddr = tokens["remote-addr"](req, res);
   const method = tokens.method(req, res);
   const status = parseInt(tokens.status(req, res), 10);
-  const url = new URL(tokens.url(req, res), "https://respec.org/");
-  const referrer = tokens.referrer(req, res);
+  /** @type {URL|null} */
+  const url = $try(() => new URL(tokens.url(req, res), "https://respec.org/"));
```

Comment:
I.... ok... I wonder if this should just be URL specific tho... a generic $try seems like trouble when we should just try/catch things.

Reply:
`tryURL(string)`?